src/osd/OSD.cc

```diff
@@ -4771,6 +4771,17 @@ void OSD::tick_without_osd_lock()
       send_failures();
     }
     map_lock.put_read();
+
+    epoch_t max_waiting_epoch = 0;
+    for (auto s : shards) {
+      max_waiting_epoch = std::max(max_waiting_epoch,
+				   s->get_max_waiting_epoch());
+    }
+    if (max_waiting_epoch > get_osdmap()->get_epoch()) {
+      dout(20) << __func__ << " max_waiting_epoch " << max_waiting_epoch
+	       << ", requesting new map" << dendl;
+      osdmap_subscribe(superblock.newest_map + 1, false);
```

Comment:
This was a bit of a red herring... iirc the problem was the teuth test that had some mon thrashing or error injection that made the mon behavior to share the new osdmap with a random osd get lost.  No osds learned about the new map and the pg create got stuck.

This only happens when there is an effectively idle cluster.  The difference between current master is that the old handle_pg_create has/had a wait_for_osdmap that did an osdmap subscribe and the new code just dumps it in the queue and doesn't explicitly subscribe.  I think pg create is the odd man out since osds regularly share epochs via ping, and this is the only way we'd get an item depending on an epoch that no osd has.